docs/slide-in-use-cases.md

```diff
@@ -0,0 +1,64 @@
+# Overview through Slide In use cases
+
+It turned out that the sliding-in feature is more complex than it had appeared in the first place.
+This paper is an overview through the use cases.
+It should help to implement and understand the feature.
+
+## Slide In New Branch Dialog
+
+The picture below shows the **Slide In New Branch Dialog**.
+This is a regular **Create New Branch Dialog** with a changed title.
+
+![Slide In New Branch Dialog](slide-in-new-branch-dialog.png "Slide In New Branch Dialog")
+
+Note the checkbox options:
+- Checkout branch
+- Overwrite existing branch
+
+## Use cases
+
+| UC ID | Branch entry exists | Local branch exists | Estimated frequency |
+| :---: | :---: | :---: | :---: |
+| 1 | false | false | ~ 75% |
+| 2 | false | true | ~ 25% |
+| 3 | true | true | < 1% |
+| 4 | true | false | < 1% |
+
+## Slide In - Logic
+
+- When the **entry does not exist**, it must be added (slid in) under the given branch.
+- When the **entry exists under the given parent**, branch layout must not be affected.
+- When the **entry exists under other parent**, it must be first removed (slid out),
+and then added (slid in) under the given parent.
```

Comment:
In terms of the user-facing naming: I think the user-facing dialog should be titled sth like `Add new branch or reattach existing branch under <parent-branch-name>`, the current `Slide In New Branch` is somewhat confusing wrt. those two distinct behaviors

Reply:
There are some factors here.
- we do not know whether the inserted branch name will be the name of an existing branch or a new one
- we shall care about consistency with the existing within IntelliJ `GitNewBranchDialog`
  - the most common case is creating a new branch 
  - it provides additional options (checkout, overwrite, (and in the near future) setTracking)

The dialog title can be changed easily to whatever we want. I am opting for `Slide In Branch Below <parent-name>`. 

Note the text field label `New branch name:` which references the `new`. Sadly, this one cannot be changed without reimplementation of the dialog.